src/stcal/ramp_fitting/likely_fit.py

```diff
@@ -62,22 +47,48 @@ def likely_ramp_fit(ramp_data, readnoise_2d, gain_2d):
 
     readtimes = get_readtimes(ramp_data)
 
+    # Check for a zeroframe
+    if ramp_data.zeroframe is not None:
+        # Read time for the zeroframe is the first entry in the first group
+        rt_zero = readtimes[0][0]
+
+        # Remove the zero readtime from the first group
+        readtimes[0] = readtimes[0][1:]
+
+        # Add the zero readtime as a separate entry in the beginning of the array
+        readtimes.insert(0, np.array([rt_zero]))
+
     covar = Covar(readtimes)
     integ_class = IntegInfo(nints, nrows, ncols)
 
     readnoise_2d = readnoise_2d / SQRT2
 
     for integ in range(nints):
-        data = ramp_data.data[integ, :, :, :]
-        gdq = ramp_data.groupdq[integ, :, :, :].copy()
-        pdq = ramp_data.pixeldq[:, :].copy()
+        data = ramp_data.data[integ]
+        gdq = ramp_data.groupdq[integ].copy()
+        pdq = ramp_data.pixeldq.copy()
+
+        # Use the zeroframe if available
+        if ramp_data.zeroframe is not None:
+            # Zeroframe image for this integration (2D)
+            zf = ramp_data.zeroframe[integ]
+
+            # Borrow the DQ image for the first group
+            zfdq = gdq[0].copy()
+
+            # Subtract the zeroframe data from the first group
+            first_group = (data[0] * ramp_data.nframes - zf) / (ramp_data.nframes - 1)
```

Comment:
Should we explicitly catch the case where `nframes=1`?  E.g., 
`if ramp_data.nframes == 1:`
`    raise ValueError("Not expecting the zero frame to be separated when there is one frame per group.")`
This should never happen, so it's me being a little paranoid.  Without the catch the code results in division by zero.